FueledUtils/ReactiveSwift/AnyAction.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright © 2020, Fueled Digital Media, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FueledUtils
+import ReactiveSwift
+
+///
+/// A type-erased Action that allows to store any `ReactiveActionProtocol`
+/// (loosing any type information at the same time)
+///
+final class AnyAction: ReactiveActionProtocol {
+	private let applyClosure: (Input) -> SignalProducer<Any, Error>
+	private let deinitToken: Lifetime.Token
+
+	init<Action: ReactiveActionProtocol>(_ action: Action) {
+		self.isEnabled = action.isEnabled
+		self.isExecuting = action.isExecuting
+		self.applyClosure = { action.apply($0 as! Action.Input).map { $0 }.mapError { $0 } }
+		self.events = action.events.map { $0.map { $0 }.mapError { $0 } }
+		self.values = action.values.map { $0 }
+		self.errors = action.errors.map { $0 }
```

Comment:
can we write 
```
self.values = action.values
self.errors = action.errors
```

is .map { $0 } required ?

Reply:
@harishsaini it is, to convert to `Any` or `Error` through type inference. I could have added `as Any` in the `map`s and `as Error` in the `mapError`, but I figured it wasn't necessary as it's a common-ish pattern in Swift